airgun/entities/report_template.py

```diff
@@ -55,6 +56,26 @@ def unlock(self, entity_name):
         view.flash.assert_no_error()
         view.flash.dismiss()
 
+    def export(self, entity_name):
+        """Export report template.
+
+         :return str: path to saved file
+        """
+        view = self.navigate_to(self, 'All')
+        view.search(entity_name)
+        view.table.row(name=entity_name)['Actions'].widget.fill('Export')
+        return self.browser.save_downloaded_file()
+
+    def generate(self, entity_name, values={}):
+        """Generate report template
+
+         :return str: path to saved file
+        """
+        view = self.navigate_to(self, 'Generate', entity_name=entity_name)
+        view.fill(values)
+        view.submit.click()
```

Comment:
is there errors popups in case of error after submitting, validation errors ..., if there is we should check them before return.

Reply:
@ldjebran , @oshtaier suggested the exact opposite here: https://github.com/SatelliteQE/airgun/pull/324#pullrequestreview-242786044
But yes, there are checks in the form, for example for e-mail format. Can we get to some agreement on what should be done here?

Reply:
Actually, there is a difference because in 6.6, the report is not downloaded straight away (as it used to in 6.5) but you are redirected to the download page. I suppose that means it IS possible to check for the pop-up.

Reply:
Yes, we should remove for 'Export' as there is no validation performed here and you get to dialog right away, but for generate, I can see validation, but it is not a pop - up:
![image](https://user-images.githubusercontent.com/10895065/58871177-39a2d800-86ca-11e9-950d-937818643dfb.png)

so not sure what validation we can add, maybe a custom one. Anyway, I feel `save_downloaded_file` should produce an exception in case dialog is absent. We can test that. If yes, we do not need any validation

Reply:
we can try ValidationErrors from widgets module? to stick with minimum validation errors, as may be useful also for negative tests, and to minimize dealing with the errors when passing to download as they may be very specific.
@oshtaier what do you think?  
anyway I am OK for the positive test cases to crash on download in case of errors and confirm ACK.

Reply:
I am always like to go with better errors, but if it at least crash on negative scenario, then I am ACK too

Reply:
Is ``view.flash.assert_no_error()`` what we are looking for?

Reply:
I believe no as we have no `flash` here. Flash errors are ones that appear in top right corner. Screenshot above is for element which directly present in DOM and will not be caught by existing validators. Anyway, I would be happy just see that scenario crash. Exact error is lower priority for me

Reply:
Your screenshot is what happens on success. However, failure looks like this:

![flash](https://user-images.githubusercontent.com/6850667/58878784-5f34df00-86d4-11e9-9a3c-8d1dfc90e602.png)

I believe that IS flash error, isn't it?

Reply:
yeap, that one will be caught =)

Reply:
so please put that line in the method

Reply:
I did. Now, when I fill in e-mail in incorrect format, the result is:
```
E               AssertionError: assert_no_error: error: 'Could not generate the report, check the form for error messages'
```